components/clarinet-format/src/formatter/mod.rs

```diff
@@ -1437,6 +1484,19 @@ mod tests_formatter {
         );
     }
     #[test]
+    fn test_preserve_newlines_inner_function() {
+        let src = r#"(define-public (increment)
+  (begin
+    (try! (stx-transfer? (var-get cost) tx-sender (var-get contract-owner)))
+
+    (ok (var-set count (+ (var-get count) u1)))
+  )
+)
+"#;
```

Comment:
We've started using the `indoc` crate in some of the other components to keep multiline readable and indented with the rest of the code:

```suggestion
        let src = indoc!(r#"
          define-public (increment)
            (begin
              (try! (stx-transfer? (var-get cost) tx-sender (var-get contract-owner)))
              (ok (var-set count (+ (var-get count) u1)))
            )
          )"#;
```

You can add the workspace version in Cargo.toml:

```toml
indoc = { workspace = true }
```

Reply:
Made it a separate issue for now. 
https://github.com/hirosystems/clarinet/issues/1891